app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt

```diff
@@ -31,17 +31,37 @@ class WelcomeFragment : Fragment() {
     private lateinit var rootView: View
     private val geoLocationViewModel by viewModel<GeoLocationViewModel>()
     val preference = Preference()
+    private lateinit var editor: SharedPreferences.Editor
+
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         rootView = inflater.inflate(R.layout.fragment_welcome, container, false)
         val thisActivity = activity
         if (thisActivity is AppCompatActivity)
             thisActivity.supportActionBar?.hide()
+
         rootView.pickCityButton.setOnClickListener {
             Navigation.findNavController(rootView)
                 .navigate(WelcomeFragmentDirections.actionWelcomeToSearch(WELCOME_FRAGMENT))
         }
 
+
+        when (preference.getBoolean("dialog", false)) {
```

Comment:
Pick better preference name. `dialog` is too generic

Reply:
The preference name changed to dialog_prompt